ddcommon/src/rate_limiter.rs

```diff
@@ -0,0 +1,186 @@
+// Copyright 2021-Present Datadog, Inc. https://www.datadoghq.com/
+// SPDX-License-Identifier: Apache-2.0
+
+use std::sync::atomic::{AtomicI64, AtomicU32, AtomicU64, Ordering};
+
+pub trait Limiter {
+    /// Takes the limit per interval.
+    /// Returns false if the limit is exceeded, otherwise true.
+    fn inc(&self, limit: u32) -> bool;
+    /// Returns the effective rate per interval.
+    /// Note: The rate is only guaranteed to be accurate immediately after a call to inc().
+    fn rate(&self) -> f64;
+}
+
+/// A thread-safe limiter built on Atomics.
+/// It's base unit is in seconds, i.e. the minimum allowed rate is 1 per second.
+/// Internally the limiter works with the system time granularity, i.e. nanoseconds on unix and
+/// milliseconds on windows.
+/// The implementation is a sliding window: every time the limiter is increased, the as much time as
+/// has passed is also refilled.
```

Comment:
Some words are missing.
Is this correct: `The implementation is a sliding window: every time the limiter is increased, the amount of time that has passed is also refilled.`?

Reply:
Thanks :-)